util/src/main/java/io/kubernetes/client/Logs.java

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client;
+
+import io.kubernetes.client.Configuration;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.util.WebSockets;
+import io.kubernetes.client.util.WebSocketStreamHandler;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.io.Reader;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.Response;
+
+/**
+ * Utility class offering streaming access to Pod logs.
+ */
+public class Logs {
+    private ApiClient apiClient;
```

Comment:
suggestion: I would make apiClient `public final` and remove setter and getter for it.

Reply:
Why `public`?

Reply:
That means you don't need a getter for it. It is a common practice for these type of variables but now that I think more about it, maybe having getter is not a bad idea. But I still suggest we don't have a setter for this and make it `private final`.

Reply:
Yeah was just curious since the rest of the generated code uses getters in favor of `public final`. Agreed that a setter probably shouldn't exist.

Reply:
Removed the setter.